common/ManagementTestShared/Redesign/MockTestEnvironment.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+using System;
+
+namespace Azure.ResourceManager.TestFramework
+{
+    public class MockTestEnvironment : TestEnvironment
+    {
+        public MockTestEnvironment() : base()
+        {
+            Environment.SetEnvironmentVariable("SUBSCRIPTION_ID", "00000000-0000-0000-0000-000000000000");
+            Environment.SetEnvironmentVariable("RESOURCE_MANAGER_URL", MockEndPoint);
+        }
+
+        public virtual string MockEndPoint => $"https://localhost:8441";
+
+        public override TokenCredential Credential
+        {
+            get
```

Comment:
I think we can just always return MockCredential here?  If so then we don't need to make the field on TestEnvironment protected internal

Reply:
make sense

Reply:
And I searched the usage of private member `_credential` in `TestEnvironment` - only one place we assigned it to null in `SetRecording`. Therefore I suppose it is valid if we just return new MockCredential here.